File: 400/423 - Reconstruct Original Digits from English.c
char* originalDigits(char* s) {
    int nums[10], letters[26] = {0}, i, j, size, count = 0;
    char *ret;
    
    for (i = 0; s[i]; ++i)
        letters[s[i] - 'a']++;
    
    nums[0] = letters['z' - 'a'];
    nums[2] = letters['w' - 'a'];
    nums[4] = letters['u' - 'a'];
    nums[6] = letters['x' - 'a'];
    nums[8] = letters['g' - 'a'];
    nums[1] = letters['o' - 'a'] - nums[0] - nums[2] - nums[4];
    nums[3] = letters['h' - 'a'] - nums[8];
    nums[5] = letters['f' - 'a'] - nums[4];
    nums[7] = letters['v' - 'a'] - nums[5];
    nums[9] = letters['i' - 'a'] - nums[5] - nums[6] - nums[8];
    
    for (i = 0; i < 10; ++i)
        size += nums[i];
    ret = (char *)malloc((size + 1) * sizeof(char));
    
    for (i = 0; i < 10; ++i) {
        for (j = 0; j < nums[i]; ++j)
            ret[count++] = i + '0';
    }
    ret[count] = '\0';
    
    return ret;
}

File: csharp/423.reconstruct-original-digits-from-english.cs
/*
 * @lc app=leetcode id=423 lang=csharp
 *
 * [423] Reconstruct Original Digits from English
 */

using System.Linq;

public class Solution {
    public string OriginalDigits(string s) {
        int[] letters = new int[26], count = new int[10];
        foreach (char c in s) {
            ++letters[(int) c - (int) 'a'];
        }
        count[6] = letters[(int) 'x' - (int) 'a'];
        count[8] = letters[(int) 'g' - (int) 'a'];
        count[2] = letters[(int) 'w' - (int) 'a'];
        count[4] = letters[(int) 'u' - (int) 'a'];
        count[0] = letters[(int) 'z' - (int) 'a'];
        count[3] = letters[(int) 'h' - (int) 'a'] - count[8];
        count[5] = letters[(int) 'f' - (int) 'a'] - count[4];
        count[7] = letters[(int) 'v' - (int) 'a'] - count[5];
        count[1] = letters[(int) 'o' - (int) 'a'] - count[0] - count[2] - count[4];
        count[9] = letters[(int) 'i' - (int) 'a'] - count[5] - count[6] - count[8];
        return new string(count.Select((n, i) => Enumerable.Repeat((char) (i + (int) '0'), n)).Aggregate((acc, list) => acc.Concat(list)).ToArray());
    }
}

